app/main.py

```diff
@@ -1,8 +1,27 @@
 class Person:
-    # write your code here
-    pass
+    people = {}
 
+    def __init__(self, name: str, age: int) -> None:
+        self.name = name
+        self.age = age
+        Person.people[name] = self
 
-def create_person_list(people: list) -> list:
-    # write your code here
-    pass
+
+def create_person_list(people_list: list) -> list:
+    person_instances = []
+
+    for person_dict in people_list:
+        person_instance = Person(person_dict["name"], person_dict["age"])
+        person_instances.append(person_instance)
+
+        wife_name = person_dict.get("wife")
+        husband_name = person_dict.get("husband")
+
+        if wife_name and wife_name in Person.people:
```

Comment:
You can replace `wife_name in Person.people` with `Person.people.get(wife_name)` and it will give you the best performance - first operation has complexity of O(n) and second O(1) - best possible performance